### PR TITLE
Align package ID validation with the client

### PR DESCRIPTION
**Pull request by scottbommarito**
See https://github.com/NuGet/NuGetGallery/pull/3285 (had to reopen because it was opened on master).

### CODE REVIEW

**Comment by scottbommarito** [comment]
Turns out we need a database migration for this change. (The max package id length changed from 128 to 100 on a couple tables.)

**Comment by scottbommarito** [comment]
@shishirx34 @skofman1 @ryuyu @xavierdecoster

File: src/NuGetGallery/Services/PackageService.cs [hunk 1]
```diff
@@ -626,13 +626,13 @@ private static void ValidateNuGetPackageMetadata(PackageMetadata packageMetadata
...
+                    // NuGet.Core compatibility - dependency package id cannot be more than PackageIdValidator.MaxPackageIdLength characters long!
+                    if (dependency.Id != null && dependency.Id.Length > PackageIdValidator.MaxPackageIdLength)
                     {
-                        throw new EntityException(Strings.NuGetPackagePropertyTooLong, "Dependency.Id", CoreConstants.MaxPackageIdLength);
+                        throw new EntityException(Strings.NuGetPackagePropertyTooLong, "Dependency.Id", PackageIdValidator.MaxPackageIdLength);
                     }
 
-                    // NuGet.Core compatibility - dependency versionspec can not be > 256 characters
+                    // NuGet.Core compatibility - dependency versionspec cannot be more than 256 characters long!
                     if (dependency.VersionRange != null && dependency.VersionRange.ToString().Length > 256)
```
Comment (xavierdecoster) [inline src/NuGetGallery/Services/PackageService.cs]: Is there a similar constant defined in one of the NuGet.* packages?

File: src/NuGetGallery/Web.config [hunk 2]
```diff
@@ -1,535 +1,535 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (xavierdecoster) [inline src/NuGetGallery/Web.config]: Think we need to align our spacing/formatting settings in the repo :) Are we all using ReSharper, or is there a way to do this with VS editorconfig or something?

Did anything change in this file besides alignment?

Comment (scottbommarito) [inline src/NuGetGallery/Web.config]: Yea I honestly can't find anything different between the two files...even the spacing seems consistent.

File: src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class MaxPackageId100 : DbMigration
+    {
+        public override void Up()
+        {
+            AlterColumn("dbo.PackageRegistrations", "Id", c => c.String(nullable: false, maxLength: 100));
+            AlterColumn("dbo.PackageDependencies", "Id", c => c.String(maxLength: 100));
+            AlterColumn("dbo.PackageLicenses", "Name", c => c.String(nullable: false, maxLength: 100));
```
Comment (scottbommarito) [inline src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs]: (it was using the same constant but was very unrelated)

File: src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+namespace NuGetGallery.Migrations
```
Comment (shishirx34) [inline src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs]: are you going to remove this migration?

File: src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        public override void Up()
+        {
+            AlterColumn("dbo.PackageRegistrations", "Id", c => c.String(nullable: false, maxLength: 100));
+            AlterColumn("dbo.PackageDependencies", "Id", c => c.String(maxLength: 100));
+            AlterColumn("dbo.PackageLicenses", "Name", c => c.String(nullable: false, maxLength: 100));
+        }
+        
+        public override void Down()
+        {
+            AlterColumn("dbo.PackageLicenses", "Name", c => c.String(nullable: false, maxLength: 128));
```
Comment (shishirx34) [inline src/NuGetGallery/Migrations/201610220037001_MaxPackageId100.cs]: remove this too?

File: src/NuGetGallery/Web.config (hunk 2 above)
Comment (shishirx34) [inline src/NuGetGallery/Web.config]: Its pretty weird, meanwhile use `?w=1` to ignore white spaces.

**Comment by scottbommarito** [comment]
Removed the migration as discussed during standup.

File: src/NuGetGallery/Migrations/201609092252096_AddIndexCredentialExpires.resx [hunk 6]
```diff
@@ -1,126 +1,126 @@
...
-  </data>
-  <data name="DefaultSchema" xml:space="preserve">
-    <value>dbo</value>
-  </data>
+﻿<?xml version="1.0" encoding="utf-8"?>
+<root>
+  <!-- 
+    Microsoft ResX Schema
```
Comment (scottbommarito) [inline src/NuGetGallery/Migrations/201609092252096_AddIndexCredentialExpires.resx]: (was an artifact of merging in master to fix merge conflicts, no idea why github went insane)

File: src/NuGetGallery/Web.config (hunk 2 above)
Comment (xavierdecoster) [inline src/NuGetGallery/Web.config]: Looking at the little `-` in front of `<?xml`, maybe an UTF8 byte was removed in this commit (good!).

**Pull request merged**